packages/cli/src/sso/saml/samlHelpers.ts

```diff
@@ -0,0 +1,133 @@
+import config from '@/config';
+import * as Db from '@/Db';
+import { AuthIdentity } from '../../databases/entities/AuthIdentity';
+import { User } from '../../databases/entities/User';
+import { getLicense } from '../../License';
+import { AuthError } from '../../ResponseHelper';
+import { hashPassword, isUserManagementEnabled } from '../../UserManagement/UserManagementHelper';
+import type { SamlPreferences } from './types/samlPreferences';
+import type { SamlUserAttributes } from './types/samlUserAttributes';
+import type { FlowResult } from 'samlify/types/src/flow';
+import type { SamlAttributeMapping } from './types/samlAttributeMapping';
+/**
+ *  Check whether the SAML feature is licensed and enabled in the instance
+ */
+export function isSamlEnabled(): boolean {
+	return config.getEnv('sso.saml.enabled');
+}
+
+export function isSamlLicensed(): boolean {
+	const license = getLicense();
+	return isUserManagementEnabled() && license.isSamlEnabled();
+}
+
+export const isSamlPreferences = (candidate: unknown): candidate is SamlPreferences => {
+	const o = candidate as SamlPreferences;
+	return typeof o === 'object' && typeof o.metadata === 'string' && typeof o.mapping === 'object';
+};
+
+export function generatePassword(): string {
+	const length = 18;
+	const charset = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789';
+	const charsetNoNumbers = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ';
+	const randomNumber = Math.floor(Math.random());
```

Comment:
This will always yield 0, is this expected?

Reply:
lol, no, good catch